diff --git a/fesvr/device.h b/fesvr/device.h
index 1387b745..0db784b6 100644
--- a/fesvr/device.h
+++ b/fesvr/device.h
@@ -4,6 +4,7 @@
 #include <vector>
 #include <queue>
 #include <cstring>
+#include <cstdint>
 #include <string>
 #include <functional>
 
diff --git a/spike_main/spike_device.cc b/spike_main/spike_device.cc
index 22143a15..eb15001f 100644
--- a/spike_main/spike_device.cc
+++ b/spike_main/spike_device.cc
@@ -198,7 +198,11 @@ spike_device::spike_device():sim(NULL),buffer(),buffer_data(){
 };
 
 spike_device::~spike_device(){
-  delete sim;delete[] srcfilename,logfilename;
+  if(sim != nullptr) {
+    delete sim;
+    sim = nullptr;
+  }
+  delete[] srcfilename,logfilename;
   for (auto& mem : buffer_data)
     if(mem.second!=nullptr) {delete mem.second;mem.second=nullptr;}
   const_buffer.clear();
@@ -669,6 +673,7 @@ int spike_device::run(meta_data* knl_data,uint64_t knl_start_pc){
   //    log_path = log_name;
 
       delete sim;
+      sim = nullptr;
   }
 
   for (auto& plugin_device : plugin_devices)
